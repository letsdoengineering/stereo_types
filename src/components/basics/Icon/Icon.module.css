.icon-size-XS {
    font-size: 0.875rem;
}

.icon-size-S {
    font-size: 1rem;
}

.icon-size-M {
    font-size: 1.25rem;
}

.icon-size-L {
    font-size: 1.5rem;
}

.icon-size-XL {
    font-size: 2.75rem;
}

.icon-color-blue {
    color: var(--light-blue)
}

.icon-color-grey {
    color: var(--dark-gray)
}
