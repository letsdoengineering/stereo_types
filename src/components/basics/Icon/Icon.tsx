import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import iconStyles from './Icon.module.css'

export type IconProps = React.AllHTMLAttributes<HTMLSpanElement> & {
  /** string that represents a specific icon */
  icon:
    | 'search'
    | 'calendar-day'
    | 'chevron-down'
    | 'chevron-up'
    | 'chevron-left'
    | 'chevron-right'
    | 'info-circle'
    | 'moon'
    | 'map-marker-alt'
    | 'ship'
    | 'times'
    | 'exclamation-triangle'
  /** string that represents the size of the icon - applied from css */
  iconSize?: ElementSize
  /** string that represents the color of the icon applied as css*/
  iconColor?: 'Blue' | 'Grey' | 'Red'
}

/** Icon: renders a span with a class-name that applies a specific icon from the fa stylesheet */
const Icon: React.FC<IconProps> = ({
  icon,
  iconSize = 'L',
  iconColor = 'Blue',
  className,
}: IconProps) => {
  const sizeClass = iconStyles[`iconSize${iconSize}`]
  const colorClass = iconStyles[`iconColor${iconColor}`]
  const classNames = getClassNames({
    defaultClasses: ['fas', `fa-${icon}`, sizeClass, colorClass],
    className,
  })

  return <span aria-hidden='true' className={classNames} />
}

export default Icon
