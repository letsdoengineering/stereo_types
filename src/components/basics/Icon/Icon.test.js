import * as React from 'react'
import renderer from 'react-test-renderer'
import Icon from './Icon'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'

describe('Icon', () => {
  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<Icon icon='CALENDAR' />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<Icon icon='CALENDAR' />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
