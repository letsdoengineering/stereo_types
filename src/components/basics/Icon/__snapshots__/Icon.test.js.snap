// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon has not changed the JSX layout of the page from previous snapshot 1`] = `
<span
  aria-hidden="true"
  className="fas fa-CALENDAR iconSizeL iconColorBlue"
/>
`;
